### PR TITLE
refactor: improve state operator typings

### PR DESCRIPTION
**Pull request by markwhitfeld**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [X] The commit message follows our guidelines: https://github.com/ngxs/store/blob/master/CONTRIBUTING.md#commit
- [X] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[X] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?

The current typings are not explicit

## What is the new behavior?
To add explicit return type typings for the state operators.

## Does this PR introduce a breaking change?
```
[ ] Yes
[X] No
```

### CODE REVIEW

**Comment by markwhitfeld** [comment]
PS. This may be step one of many to get the TS 3.4 issues resolved.

**Comment by markwhitfeld** [comment]
Also, the commented out sections in the dtslint test files are existing typings issues that need to be solved.

**Comment by markwhitfeld** [comment]
Ok, These CI errors give a great starting point to get this all fixed!

**Comment by splincode** [comment]
Good work!

File: packages/store/operators/src/insert-item.ts [hunk 1]
```diff
@@ -1,15 +1,16 @@
 import { isNil } from './utils';
+import { StateOperator } from '@ngxs/store/src/symbols';
```
Comment (arturovt) [inline packages/store/operators/src/insert-item.ts]: Fix this import please

**Comment by splincode** [comment]
@arturovt Could you help with this pull request?

**Comment by arturovt** [comment]
I'm very busy on the job, I can have a look on friday or weekends.

**Comment by arturovt** [comment]
Mark, I couldn't push to your repo, so I pushed to the main repo.
This PR will now be tracked in PR #1128 
Closing this one

**Pull request closed without merging**